website/snippets/_render-method.md

```diff
@@ -0,0 +1,29 @@
+#### The render method
+
+The `.render()` method is generally used to resolve or evaluate Jinja expressions (such as `{{ source(...) }}`) during runtime. When a pre-hook or post-hook contains a dynamic reference, such as a table or column, dbt might not automatically resolve the reference correctly, particularly when certain flags (such as `--empty`) are applied.
+
+When using the --empty flag, dbt may skip processing `ref()` or `source()` for optimization. To avoid compilation errors and to explicitly tell dbt to process a specific relation (`ref()` or `source()`), use the `.render()` method in your model file. For example:
+
+
+<File name='models.sql'>
+
+```Jinja
+
+-- models/staging/stg_sys__customers.sql
```

Comment:
i don't think you need all this code @nataliefiann , esp as its specific to the user's code. i think just having:

`{{ config(
    pre_hook = [
        "alter external table {{ source('stripe', 'customers').render() }} refresh"
    ]`
    
    is enough